internal/route/api/v1/repo/file.go

```diff
@@ -45,6 +48,72 @@ func GetArchive(c *context.APIContext) {
 	repo.Download(c.Context)
 }
 
+func GetRepoTree(c *context.APIContext) {
+	repoPath := db.RepoPath(c.Params(":username"), c.Params(":reponame"))
+	gitRepo, err := git.OpenRepository(repoPath)
+
+	if err != nil {
+		c.ServerError("OpenRepository", err)
+		return
+	}
+	c.Repo.GitRepo = gitRepo
```

Comment:
`c.Repo.GitRepo` is already set in `context.RepoRef()` call, so you can just use it.

Reply:
Thanks, fixed it.